### PR TITLE
Add subtree metadata to implicit tiling

### LINKED ISSUE #10088: Add a class for subtree metadata in implicit tiling
For implicit metadata, the latest spec revisions call for adding metadata per subtree. 

This will be JSON-based metadata properties, so it will work very similarly to `TilesetMetadata` and `GroupMetadata`. These use the static methods from `MetadataEntity` to implement the metadata, so most of the low-level details are abstracted.

This involves:

- [x] Create a class `Scene/ImplicitSubtreeMetadata.js` similar to `TilesetMetadata` and `GroupMetadata`. see here for the [latest schema](https://github.com/CesiumGS/3d-tiles/blob/subtree-extension-folding/extensions/3DTILES_implicit_tiling/schema/subtree/subtreeMetadata.schema.json)
- [x] Update `ImplicitSubtree` to create an `ImplicitSubtreeMetadata` class. 
- [x] Update `Cesium3DTileFeature.getPropertyInherited()` to add a clause for searching the subtree metadata. This should access it like `tile.implicitSubtree.metadata` (if `tile.implicitSubtree` is defined). This clause should come after tile metadata but before the group metadata since the precedence is from most specific to most general.
- [x] update unit tests. Hopefully with the changes from #10084 it'll be easier to modify sample data to add subtree metadata.

CC @j9liu

### PR DESCRIPTION
**Pull request by j9liu**
Resolves #10088.

This PR adds a subtree metadata class, `ImplicitSubtreeMetadata`, and implements the ability to get properties from it. The unit tests were updated accordingly with the new sample data (thanks @ptrgags for creating it!).

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @j9liu!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: Source/Scene/ImplicitSubtree.js [hunk 1]
```diff
@@ -64,6 +65,7 @@ export default function ImplicitSubtree(
   this._readyPromise = when.defer();
 
   // properties for 3DTILES_metadata
+  this._metadata = undefined;
   this._metadataTable = undefined;
   this._metadataExtension = undefined;
```
Comment (ptrgags) [inline Source/Scene/ImplicitSubtree.js:70]: Hm... while we're at it, could we rename `metadataTable -> tileMetadataTable` and `metadataExtension -> tileMetadataExtension`? This is more explicit, and will make things easier to understand in a future PR when you'll be updating tile metadata and adding content metadata.

**Review by ptrgags — COMMENTED** [review]
@j9liu looks solid overall! noticed a couple of small refactorings, and a couple minor code style things.

File: Specs/Data/Cesium3DTiles/Metadata/ImplicitSubtreeMetadata/tileset.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      ]
+    },
+    "content": {
+      "uri": "content/{level}/{x}/{y}.b3dm"
+    },
+    "extensions": {
+      "3DTILES_implicit_tiling": {
+        "subdivisionScheme": "QUADTREE",
+        "subtreeLevels": 2,
+        "maximumLevel": 1,
```
Comment (ptrgags) [inline Specs/Data/Cesium3DTiles/Metadata/ImplicitSubtreeMetadata/tileset.json:66]: once #10094 is merged, remember to update this to use `availableLevels`, thanks!

**Comment by j9liu** [comment]
@ptrgags - just pushed some fixes, let me know if anything else comes up!

File: CHANGES.md [hunk 3]
```diff
@@ -9,6 +9,8 @@
 - Added `depthPlaneEllipsoidOffset` to Viewer and Scene constructors to address rendering artefacts below ellipsoid zero elevation. [#9200](https://github.com/CesiumGS/cesium/pull/9200)
 - Added support for `debugColorTiles` in `ModelExperimental`. [#10071](https://github.com/CesiumGS/cesium/pull/10071)
 - Added support for shadows in `ModelExperimental`. [#10077](https://github.com/CesiumGS/cesium/pull/10077)
+- Added the ability to load implicit tiling subtrees using JSON files. [#10086](https://github.com/CesiumGS/cesium/pull/10086)
```
Comment (ptrgags) [inline CHANGES.md:12]:
```suggestion
- Added the ability to load implicit tiling subtree files from JSON. [#10086](https://github.com/CesiumGS/cesium/pull/10086)
```

**Review by ptrgags — APPROVED** [review]
@j9liu looks good, thanks! One tiny wording tweak, but I'll commit that through the GH UI and merge this in a moment.

**Pull request merged**